### ISSUE TITLE
Container keep on restarting with /merged/vault/tls/ca.crt not found

### ISSUE THREAD

**Issue by dhileepbalaji** [bank-vaults/bank-vaults#1187]
> Bugs should be filed for issues encountered whilst operating bank-vaults.
> You should first attempt to resolve your issues through the community support
> channels, e.g. Slack, in order to rule out individual configuration errors.
> Please provide as much detail as possible.

**Describe the bug**:
When Application container restarts sometimes it keeps on restarting with below error.

        message: 'OCI runtime create failed: container_linux.go:349: starting container
          process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58:
          mounting \\\"/var/lib/kubelet/pods/8f10a795-f642-4aa7-92f9-8f6cc830f418/volume-subpaths/vault-tls/encorgstatusschtsk/2\\\"
          to rootfs \\\"/var/lib/docker/overlay2/8dfa71036ba33e1aab45172f7443cb175ff0b8e2cd6aa2c3f80074c4c41d7721/merged\\\"
          at \\\"/var/lib/docker/overlay2/8dfa71036ba33e1aab45172f7443cb175ff0b8e2cd6aa2c3f80074c4c41d7721/merged/vault/tls/ca.crt\\\"

**Expected behaviour**:
Application Container should restart without any errors.

**Steps to reproduce the bug**:
Configure the application to use vaults secrets and make the application container to fail, by listening on port 80 for 2 mins and close the listening port. Configure liveliness probe for port 80 and wait for health check to fail.

**Additional context**:

**Environment details**:
- Kubernetes version (e.g. v1.10.2): **v1.16.13**
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): **AKS**
- bank-vaults version (e.g. 0.4.17):  **banzaicloud/bank-vaults:1.1.0**
- Install method (e.g. helm or static manifests): **helm**
- Logs from the misbehaving component (and any other relevant logs):
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data:
``` yaml 
apiVersion: v1
kind: Pod
metadata:
  annotations:
    cattle.io/timestamp: "2020-10-13T20:37:04Z"
    field.cattle.io/ports: '[[{"containerPort":8080,"dnsName":"external-organization-status-schtsk-uat","kind":"ClusterIP","name":"http","protocol":"TCP"}]]'
    vault.security.banzaicloud.io/vault-addr: https://k8vault-vault.vault:8200
    vault.security.banzaicloud.io/vault-tls-secret: k8vault-vault-tls
  creationTimestamp: "2020-11-07T17:23:22Z"
  generateName: external-organization-status-schtsk-uat-554b45b9cd-
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-11-07T17:23:25Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2020-11-15T09:46:23Z"
    message: 'containers with unready status: [encorgstatusschtsk]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2020-11-15T09:46:23Z"
    message: 'containers with unready status: [encorgstatusschtsk]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2020-11-07T17:23:22Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://0486966f7c5a251de5ceca1564ad6904a8eb55e442a34daef1f4148a4058a8aa
    image: [redacted]
    imageID: [redacted]
    lastState:
      terminated:
        containerID: docker://0486966f7c5a251de5ceca1564ad6904a8eb55e442a34daef1f4148a4058a8aa
        exitCode: 128
        finishedAt: "2020-11-19T05:22:04Z"
        message: 'OCI runtime create failed: container_linux.go:349: starting container
          process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58:
          mounting \\\"/var/lib/kubelet/pods/8f10a795-f642-4aa7-92f9-8f6cc830f418/volume-subpaths/vault-tls/encorgstatusschtsk/2\\\"
          to rootfs \\\"/var/lib/docker/overlay2/89c1e504c22b1300ce76c90751a0bcf10967a86867ea7e6ce834d00f81f2ba3b/merged\\\"
          at \\\"/var/lib/docker/overlay2/89c1e504c22b1300ce76c90751a0bcf10967a86867ea7e6ce834d00f81f2ba3b/merged/vault/tls/ca.crt\\\"
          caused \\\"no such file or directory\\\"\"": unknown'
        reason: ContainerCannotRun
        startedAt: "2020-11-19T05:22:04Z"
    name: encorgstatusschtsk
    ready: false
    restartCount: 1094
    started: false
    state:
      waiting:
        message: back-off 5m0s restarting failed container=encorgstatusschtsk pod=external-organization-status-schtsk-uat-554b45b9cd-gpt79_uat(8f10a795-f642-4aa7-92f9-8f6cc830f418)
        reason: CrashLoopBackOff
  hostIP: 10.14.193.106
  initContainerStatuses:
  - containerID: docker://c370fcd5b3874e23971ba4f70fac3acc3b80e53816acd80db7451dd67d96998f
    image: banzaicloud/vault-env:1.1.0
    imageID: docker-pullable://banzaicloud/vault-env@sha256:558a8311787ae930aafbd9877688f35d6165feebc3f67882d7971eb3a1cd6a05
    lastState: {}
    name: copy-vault-env
    ready: true
    restartCount: 0
    state:
      terminated:
        containerID: docker://c370fcd5b3874e23971ba4f70fac3acc3b80e53816acd80db7451dd67d96998f
        exitCode: 0
        finishedAt: "2020-11-07T17:23:25Z"
        reason: Completed
        startedAt: "2020-11-07T17:23:24Z"
  phase: Running
  podIP: 10.14.193.111
  podIPs:
  - ip: 10.14.193.111
  qosClass: Burstable
  startTime: "2020-11-07T17:23:22Z" `
```

/kind bug

**Comment by bipeens** [comment]
@dhileepbalaji , Did you find any solution or workaround for this? I am facing this issue too in my cluster. If I delete the pod then new pod comes properly but if any of the container received SIGTERM and tries to restart end up having this issue.

**Comment by oxr463** [comment]
@dhileepbalaji what was the resolution for this issue?

**Comment by oxr463** [comment]
@dhileepbalaji any updates on this?

**Comment by dhileepbalaji** [comment]
@oxr463 No, not yet. Still working on it.

**Comment by dhileepbalaji** [comment]
It seems to be resolved in 1.19
https://github.com/kubernetes/kubernetes/issues/68211

**Comment by bipeens** [comment]
Yes, This has been resolved in latest version. Currently we also fixed by upgrading bank-vaults to 1.8.0 docker image. This issue was because the vault-tls cert was getting updated for every re-install of vault-operator chart. So as it was mounted into pod with volume subpath, a restart of the container(or SIGTERM) resulted in to the issue which is explained https://github.com/kubernetes/kubernetes/issues/68211
After upgrading, the vault-tls secrets is not getting updated now even we re-install vault-operator chart.

Also as vault-tls certs are renewed once a year,  the fix https://github.com/banzaicloud/bank-vaults/pull/976 which is merged the renewals of the cert will happen smoothly and we should not face this issue
